experiment/secret-sync-controller/secret-sync-config.go

```diff
@@ -0,0 +1,161 @@
+package main
+
+// configuration and sync-pair defination
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	secretmanager "cloud.google.com/go/secretmanager/apiv1"
+	secretmanagerpb "google.golang.org/genproto/googleapis/cloud/secretmanager/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// Structs for configuration
+type SecretSyncConfig struct {
+	Specs []SecretSyncSpec `yaml:"specs"`
+}
+type SecretSyncSpec struct {
+	// the target can be either a K8s secret or a SecretManager secret
+	Source      TargetSpec `yaml:"source"`
+	Destination TargetSpec `yaml:"destination"`
+}
+type TargetSpec struct {
+	// assert that one of the two should be `nil`
+	Kubernetes    *KubernetesSpec    `yaml:"kubernetes,omitempty"`
+	SecretManager *SecretManagerSpec `yaml:"secretManager,omitempty"`
+}
+type KubernetesSpec struct {
+	Namespace string   `yaml:"namespace"`
+	Secret    string   `yaml:"secret,omitempty"`
+	DenyList  []string `yaml:"denyList,omitempty"`
```

Comment:
As we discussed this morning, I would drop the denyList stuff for now.  Instead, give KubernetesSpec an additional field of `Key string` or something to help the mapping between kubernetes secrets which expect key=value pairs, and secret manager secrets which don't.